api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -389,29 +393,20 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     permissions.setUser(user);
 
     dbWorkspace.addWorkspaceUserRole(permissions);
-
     dbWorkspace = workspaceService.getDao().save(dbWorkspace);
-    List<JSONObject> demoCohorts = cloudStorageService.readAllDemoCohorts();
-    for (JSONObject cohort: demoCohorts) {
-      // TODO: Make this a service method to avoid duplication with CohortsController
-      Cohort dbCohort = new Cohort();
-      dbCohort.setName(cohort.getString("name"));
-      dbCohort.setDescription(cohort.getString("description"));
-      dbCohort.setCriteria(cohort.get("criteria").toString());
-      dbCohort.setType(cohort.getString("type"));
-      dbCohort.setCreator(userProvider.get());
-      dbCohort.setWorkspaceId(dbWorkspace.getWorkspaceId());
-      dbCohort.setCreationTime(now);
-      dbCohort.setLastModifiedTime(now);
-      dbCohort.setVersion(1);
-      try {
-        dbCohort = cohortDao.save(dbCohort);
-      } catch (DataIntegrityViolationException e) {
-        throw new ServerErrorException(String.format(
-            "Cohort \"/%s/%s/%d\" already exists.",
-            dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getWorkspaceId(), dbCohort.getCohortId()));
-      }
-    }
+
+    org.pmiops.workbench.db.model.Workspace finalDbWorkspace = dbWorkspace;
+    cloudStorageService.readAllDemoCohorts().stream()
+            .map(apiCohort -> cohortFactory.createCohort(apiCohort, userProvider.get(), finalDbWorkspace.getWorkspaceId())
+            ).forEach(dbCohort -> {
+              try {
+                dbCohort = cohortDao.save(dbCohort);
+              } catch (DataIntegrityViolationException e) {
+                throw new ServerErrorException(String.format(
```

Comment:
Nit: This should be a conflict exception I believe. 

Edit: Although, this is about loading in demos. This may actually not be an exception that we want to block creating a workspace on, but rather log and move on. I would probably approach it that way here, rather than throwing a `ServerErrorException`

Reply:
(We may also want to do that with the other demos, but that seems out of scope for this work)

Reply:
a `BadRequestException` seems more appropriate than `ServerErrorException`. I'm not sure about removing the exception message altogether though. The changes in this file are just for refactoring so I don't want to make too many functionally visible changes.

Reply:
we could file a ticket if we find repro steps for a bug where this manifests.